daemon/dns/dns_resolvectl.go

```diff
@@ -0,0 +1,69 @@
+package dns
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/NordSecurity/nordvpn-linux/internal"
+)
+
+// Executables
+const (
+	// execResolvectl defines resolvectl executable
+	execResolvectl = "resolvectl"
+)
+
+// Systemd-resolved and resolvectl based DNS handling method
+type Resolvectl struct{}
+
+func (m *Resolvectl) Set(iface string, nameservers []string) error {
+	return setDNSWithResolvectl(iface, nameservers)
+}
+
+func (m *Resolvectl) Unset(iface string) error {
+	return unsetDNSWithResolvectl(iface)
+}
+
+func (m *Resolvectl) IsAvailable() bool {
+	return internal.IsCommandAvailable(execResolvectl)
+}
+
+func (m *Resolvectl) Name() string {
+	return "resolvectl"
+}
+
+func setDNSWithResolvectl(iface string, addresses []string) error {
```

Comment:
the [documentation](https://github.com/poettering/systemd/commit/8cedb0aef94da880e61b4c8cfeb7f450f8760ec6) suggests that following should be used to configure:
```
resolvectl domain <interface> '~.'
resolvectl default-route <interface> true
resolvectl dns <interface> 8.8.8.8
```

same here with the default-route, might be optional.

Reply:
Thx, added.